src/vs/platform/terminal/common/terminal.ts

```diff
@@ -799,6 +799,15 @@ export interface ITerminalContributions {
 	profiles?: ITerminalProfileContribution[];
 }
 
+export interface ITerminalQuickFixContribution {
+	id: string;
+	commandLineMatcher: string | RegExp;
+	outputMatcher: ITerminalOutputMatcher;
+	exitStatus?: boolean;
+	commandToRun?: string;
+	linkToOpen?: string;
```

Comment:
This will only work for dynamic links/commands if we bring capture groups in, this is the idea:

```js
{
  outputMatcher: /git (?P<command>.+)/,
  commandToRun: "git {command}"
}
```

Reply:
yes i thought that's what we're going for?

Reply:
I think we will still want to check the command line

Reply:
my thought here was 

```
commandLineMatcher: `git push`
outputMatcher: {
   commandLine: /git push --set-upstream origin (?P<branch>.+)/
}
commandToRun: "git push --set-upstream origin {branch}"
```

Reply:
yes 👍🏼